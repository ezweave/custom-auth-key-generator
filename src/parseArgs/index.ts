import { Args, ArgTypes, parseArgs } from './parseArgs';

export { Args, ArgTypes, parseArgs };
