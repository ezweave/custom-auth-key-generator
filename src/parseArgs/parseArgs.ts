import { get } from 'lodash';

export enum ArgTypes {
    configFile = 'configFile',
    headerPrefix = 'headerPrefix',
    path = 'path',
    privateKey = 'privateKey',
    publicKey = 'publicKey',
}

export interface Args {
    configFile?: string
    headerPrefix?: string,
    path: string,
    privateKey?: string,
    publicKey?: string,
}

export interface ParseArgs {
// eslint-disable-next-line @typescript-eslint/no-explicit-any 
    (args: any): Args
}

export const parseArgs: ParseArgs = (
  args 
) => ({
    configFile: get(args, ArgTypes.configFile),
    headerPrefix: get(args, ArgTypes.headerPrefix),
    path: get(args, ArgTypes.path),
    privateKey: get(args, ArgTypes.privateKey),
    publicKey: get(args, ArgTypes.publicKey),
});
