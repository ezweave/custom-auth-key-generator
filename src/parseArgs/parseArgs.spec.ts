import { parseArgs } from './parseArgs';

describe('parseArgs', () => {
    it('parses avaialable args', () => {
        const args = {
            configFile: 'something.yml',
            headerPrefix: 'something',
            path: 'foo/foo/bar',
            privateKey: 'adfadsfasdfasdf',
            publicKey: 'jljjjjjjj',
        };
        expect(
            parseArgs(args)
        ).toMatchSnapshot();
    });
});
