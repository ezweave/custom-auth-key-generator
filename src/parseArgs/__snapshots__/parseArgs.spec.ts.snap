// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseArgs parses avaialable args 1`] = `
Object {
  "configFile": "something.yml",
  "headerPrefix": "something",
  "path": "foo/foo/bar",
  "privateKey": "adfadsfasdfasdf",
  "publicKey": "jljjjjjjj",
}
`;
